pkg/collector/corechecks/system/wincrashdetect/wincrashdetect.go

```diff
@@ -0,0 +1,200 @@
+		dd_config.SystemProbe.GetString("system_probe_config.sysprobe_socket"))
+	if err != nil {
+		return wcd.handleStartupError(err)
+	}
+
+	data, err := sysProbeUtil.GetCheck(sysconfig.WindowsCrashDetectModule)
+	if err != nil {
+		return wcd.handleStartupError(err)
+	}
+	crash, ok := data.(probe.WinCrashStatus)
+	if !ok {
+		return fmt.Errorf("Raw data has incorrect type")
+	}
+	/*
+	 * originally did this with a sync.once.  The problem is the check is run prior to the
+	 * system probe being successfully started.  This is OK; we just need to detect the BSOD
+	 * on first available run.
+	 *
+	 * so keep running the check until we are able to talk to system probe, after which
+	 * we don't need to run any more
+	 */
+	wcd.hasRunOnce = true
+	if !crash.Success {
+		return fmt.Errorf("Error getting crash data %s", crash.ErrString)
+	}
+
+	if len(crash.FileName) == 0 {
+		// no crash data present.  this is actually good news.  We don't need to do anything
+		// else
+		return nil
+	}
+
+	if haveAlreadyReported(crash) {
+		log.Infof("Not reporting event on already reported crash")
+		return nil
+	}
+	sender, err := wcd.GetSender()
+	if err != nil {
+		return err
+	}
+	ev := event.Event{
+		Priority:       event.EventPriorityNormal,
+		SourceTypeName: crashDetectCheckName,
+		EventType:      crashDetectCheckName,
+		Title:          formatTitle(crash),
+		Text:           formatText(crash),
+	}
+	log.Infof("Sending event %v", ev)
+	sender.Event(ev)
+	sender.Commit()
+	setReported(crash)
+	return nil
+}
+
+var hive = registry.LOCAL_MACHINE
+var baseKey = `SOFTWARE\Datadog\Datadog Agent\windows_crash_reporting`
+
+const reportedKey = `lastReported`
+
+func generateReportedValue(wcs probe.WinCrashStatus) string {
+	return fmt.Sprintf("%s_%s", wcs.FileName, wcs.DateString)
+}
+func haveAlreadyReported(wcs probe.WinCrashStatus) bool {
+
+	newval := generateReportedValue(wcs)
+
+	k, err := registry.OpenKey(hive, baseKey, registry.QUERY_VALUE)
+	if err != nil {
+		// key not even there
+		return false
+	}
+	defer k.Close()
+	reportedval, _, err := k.GetStringValue(reportedKey)
+	if err != nil {
+		return false
+	}
+	if newval == reportedval {
+		return true
+	}
+	return false
+}
+
+func setReported(wcs probe.WinCrashStatus) {
+	newval := generateReportedValue(wcs)
+
+	k, _, err := registry.CreateKey(hive, baseKey, registry.ALL_ACCESS)
+	if err != nil {
+		// key not even there
+		return
+	}
+	defer k.Close()
+	// if we can't set the value, there's nothing we can do.  On next agent
+	// start, the same crash will be reported if the file is still there.
+	_ = k.SetStringValue(reportedKey, newval)
+}
+
+func formatTitle(c probe.WinCrashStatus) string {
+	return "A Windows system crash was detected"
+}
+
+func formatText(c probe.WinCrashStatus) string {
+	baseString := `A system crash was detected.
+	The crash occurred at %s.
+	The offending moudule is %s.
+	The bugcheck code is %s`
+	return fmt.Sprintf(baseString, c.DateString, c.Offender, c.BugCheck)
```

Comment:
Is second and subsequent lines will have extra space and it is desired?

Reply:
yes.